src/Package/Impl/Utilities/ProjectUtilities.cs

```diff
@@ -42,5 +52,95 @@ public static class ProjectUtilities {
 
             return result;
         }
+
+        public static EnvDTE.Project GetActiveProject() {
+            DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
+            var projects = (Array)dte.ActiveSolutionProjects;
+            if (projects != null && projects.Length == 1) {
+                return (EnvDTE.Project)projects.GetValue(0);
+            }
+            return null;
+        }
+
+        public static void AddNewItem(string templateName, string name, string extension, string destinationPath) {
```

Comment:
If it's labeled the same, there should be a very good reason for it to not behave the same. In this case, I simply don't see such a reason, especially given the amount of custom code that this unusual behavior requires compared to the regular behavior. And click-wise, it is not only not better, it's actually worse. For example, consider C# again. When I do Add Class, the sequence is as follows:

1. Right-click folder/project -> "Add Class"
2. Type the name of the class (focus will already be in the right field in Add New Item dialog).
3. Press Enter.

With the flow implemented here:

1. Right-click project/folder -> "Add Script"
2. Right-click on the newly added file again -> "Rename" (or click-click it slowly).
3. Type the name of the file.
4. Press Enter.

So basically this workflow _adds_ an extra two clicks in all scenarios, unless I'm okay with "rscript1.R" etc. Which is pretty much never the case outside of hello world type apps.